lib/css-converter.js

```diff
@@ -0,0 +1,303 @@
+import { CssSelectorParser } from 'css-selector-parser';
+import { escapeRegExp, find, filter, isString, isObject } from 'lodash';
+import { errors } from 'appium-base-driver';
+
+const CssConverter = {};
+
+const parser = new CssSelectorParser();
+parser.registerSelectorPseudos('has');
+parser.registerNestingOperators('>', '+', '~');
+parser.registerAttrEqualityMods('^', '$', '*', '~');
+parser.enableSubstitutes();
+
+const BOOLEAN_ATTRS = [
+  'visible', 'accessible', 'accessibility-container', 'enabled',
+];
+
+const NUMERIC_ATTRS = [
+  'index'
+];
+
+const STR_ATTRS = [
+  'label', 'name', 'value', 'type',
+];
+
+const ALL_ATTRS = [
+  ...BOOLEAN_ATTRS,
+  ...NUMERIC_ATTRS,
+  ...STR_ATTRS,
+];
+
+const ATTRIBUTE_ALIASES = [
+  ['name', ['id']],
+  ['index', ['nth-child']],
+];
+
+/**
+ * Convert hyphen separated word to snake case
+ *
+ * @param {string} str
+ * @returns {string} The hyphen separated word translated to snake case
+ */
+function toSnakeCase (str) {
```

Comment:
I thought snake-case is a string separated with underscores `_`. Hyphen-separated strings don't have a special name

Reply:
I changed it to camel case.